docs/design/safety/terminology.md

```diff
@@ -0,0 +1,195 @@
+# Safety: Terminology
+
+<!--
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+<!-- toc -->
+
+## Table of contents
+
+-   [Core Terminology](#core-terminology)
+    -   [_Hazard_](#hazard)
+    -   [_Bug_ or _defect_](#bug-or-defect)
+        -   [_Active bug_](#active-bug)
+        -   [_Latent bug_](#latent-bug)
+    -   [_Safety_](#safety)
+    -   [_Code_, _software_, or _program safety_](#code-software-or-program-safety)
+    -   [_Safety bugs_](#safety-bugs)
+    -   [_Initial bug_](#initial-bug)
+    -   [_Fail-stop_](#fail-stop)
+-   [Vulnerability Terminology](#vulnerability-terminology)
+    -   [_Vulnerability_ or _security vulnerability_](#vulnerability-or-security-vulnerability)
+    -   [_Vulnerability defense_](#vulnerability-defense)
+        -   [_Detecting_](#detecting)
+        -   [_Mitigating_](#mitigating)
+        -   [_Preventing_ vulnerabilities](#preventing-vulnerabilities)
+        -   [_Ensuring_ correctness](#ensuring-correctness)
+        -   [_Hardening_](#hardening)
+-   [Memory Safety Specifics](#memory-safety-specifics)
+    -   [_Memory safety_](#memory-safety)
+        -   [_Temporal safety_](#temporal-safety)
+        -   [_Spatial safety_](#spatial-safety)
+        -   [_Type safety_](#type-safety)
+        -   [_Initialization safety_](#initialization-safety)
+        -   [_Data-race safety_](#data-race-safety)
+    -   [_Memory safety bug_](#memory-safety-bug)
+    -   [_Memory-safe platform_ or _environment_](#memory-safe-platform-or-environment)
+    -   [_Memory-safe language_](#memory-safe-language)
+
+<!-- tocstop -->
+
+## Core Terminology
+
+### _Hazard_
+
+Unsafe coding construct that may lead to a bug or vulnerability.
+
+### _Bug_ or _defect_
+
+Reachable program behavior contrary to the author's intent.
+
+#### _Active bug_
+
+Buggy behavior that is actively occurring for users of the program.
+
+#### _Latent bug_
+
+Buggy behavior that does not currently occur for users, but is reachable.
+Behaviors that are reachable, and so _can_ happen, but don't happen today in
+practice _are always still bugs_!
+
+### _Safety_
+
+Absent a qualifier or narrow context, refers to _system safety_, and _safety
```

Comment:
"system safety" and "safety engineering" don't seem to be defined here, making it unclear what we are defining "safety" to mean.